src/components/AccountDrawer/IconButton.tsx

```diff
@@ -81,4 +92,109 @@ const IconButton = ({ Icon, ...rest }: IconButtonProps | IconLinkProps) => (
   </IconBlock>
 )
 
+type IconWithTextProps = (IconButtonProps | IconLinkProps) & {
+  text: string
+  onConfirm?: () => void
+  onShowConfirm?: (on: boolean) => void
+}
+
+const TextWrapper = styled.div`
+  display: flex;
+  flex-shrink: 0;
+  overflow: hidden;
+  min-width: min-content;
+`
+
+/**
+ * Allows for hiding and showing some text next to an IconButton
+ * Note that for width transitions to animate in CSS we need to always specify the width (no auto)
+ * so there's resize observing and measuring going on here.
+ */
+export const IconWithTextButton = ({ Icon, text, onConfirm, onShowConfirm, onClick, ...rest }: IconWithTextProps) => {
+  const [showText, _setShowText] = useState(false)
+  const frameObserver = useResizeObserver<HTMLElement>()
+  const hiddenObserver = useResizeObserver<HTMLElement>()
+
+  const setShowText = useCallback(
+    (val: boolean) => {
+      _setShowText(val)
+      onShowConfirm?.(val)
+    },
+    [onShowConfirm]
+  )
+
+  const dimensionsRef = useRef({
+    frame: 0,
+    hidden: 0,
+  })
+  const dimensions = (() => {
+    // once opened, we avoid updating it to prevent constant resize loop
+    if (!showText) {
+      dimensionsRef.current = { frame: frameObserver.width || 0, hidden: hiddenObserver.width || 0 }
+    }
+    return dimensionsRef.current
+  })()
+
+  // keyboard action to cancel
+  useEffect(() => {
+    if (!showText) return
+    const isClient = typeof window !== 'undefined'
+    if (!isClient) return
+    if (!showText) return
+    const keyHandler = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        setShowText(false)
+        e.preventDefault()
+        e.stopPropagation()
+      }
+    }
+    window.addEventListener('keydown', keyHandler, { capture: true })
+    return () => {
+      window.removeEventListener('keydown', keyHandler, { capture: true })
+    }
+  }, [setShowText, showText])
+
+  const xPad = showText ? 12 : 0
+  const width = showText ? dimensions.frame + dimensions.hidden + xPad : 32
+
+  return (
+    <IconBlock
+      ref={frameObserver.ref}
+      {...rest}
+      style={{
+        width,
+        paddingLeft: xPad,
+        paddingRight: xPad,
+      }}
```

Comment:
is there a way to pass dynamic styles to a styled component? because this one is technically a parent wrapper of a styled component but we need to send in arbitrary numerical values, but defining something like a new prop in types + in the styled definition is *so* much extra code and i don't see any upside (for dynamic styles like this anyway, as they are already nicely typed on the `style` property).